Makefile

```diff
@@ -62,25 +65,39 @@ $(PYTHON_DEPENDENCY_FILE): $(REQUIREMENTS_FILES)
 	touch $(PYTHON_DEPENDENCY_FILE)
 
 
-lint: lint-all																			## Lints the code for all available runtimes
-	@echo "CFN Lint complete"
-.PHONY: lint
+.PHONY: test
+test: lint-all-templates lint-sam-apps test-sam-apps								    ## Lints then tests code for all available runtimes
 
 
-lint-all: $(ALL_CFN_TEMPLATES)
-$(ALL_CFN_TEMPLATES):
-	cfn-lint -t $@
-.PHONY: lint-all $(ALL_CFN_TEMPLATES)
+.PHONY: lint-all-templates
+lint-all-templates: $(ALL_CFN_TEMPLATES)
+	cfn-lint -i W2001 -t $?
 
 
-test: lint test-all-apps																		## Lints then tests code for all available runtimes
-.PHONY: test
+.PHONY: lint-sam-apps
+lint-sam-apps: $(SAM_APP_LINT_RESULTS)
+$(SAM_APP_LINT_RESULTS):
+	cd $(@D) && $(MAKE) lint
+
+
+.PHONY: test-sam-apps
+test-sam-apps: $(SAM_APP_TEST_RESULTS)
+$(SAM_APP_TEST_RESULTS):
+	cd $(@D) && $(MAKE) test
+
+
+.PHONY: test-sam-apps
+package-sam-apps: $(SAM_APP_ZIPS)
+$(SAM_APP_ZIPS):
+	cd $(@D) && $(MAKE) package
 
 
-test-all-apps: $(SAM_APPS)
-$(SAM_APPS):
-	$(MAKE) test -C $@
-.PHONY: test-all-apps $(SAM_APPS)
+.PHONY: clean-sam-apps
+clean-sam-apps:
```

Comment:
NOTE: Probably a way to do this without a for loop, but punting on it for now b/c it's just the clean step